modules/inference/src/inference_generator.go

```diff
@@ -0,0 +1,306 @@
+}
+
+// ValidateConfig validates the completed inference configs are valid or not.
+func (infer *Inference) ValidateConfig() error {
+	if infer.Framework != "Ollama" && infer.Framework != "KubeRay" {
+		return errors.New("framework must be Ollama or KubeRay")
+	}
+	if infer.TopK <= 0 {
+		return errors.New("topK must be greater than 0 if exist")
+	}
+	if infer.TopP <= 0 || infer.TopP > 1 {
+		return errors.New("topP must be greater than 0 and less than or equal to 1 if exist")
+	}
+	if infer.Temperature <= 0 {
+		return errors.New("temperature must be greater than 0 if exist")
+	}
+	if infer.NumPredict < -2 {
+		return errors.New("numPredict must be greater than or equal to -2")
+	}
+	if infer.NumCtx <= 0 {
+		return errors.New("numCtx must be greater than 0 if exist")
+	}
+	return nil
+}
+
+// GenerateInferenceResource generates the Kubernetes Service related to the inference module service.
+//
+// Note that we will use the SDK provided by the kusion module framework to wrap the Kubernetes resource
+// into Kusion resource.
+func (infer *Inference) GenerateInferenceResource(request *module.GeneratorRequest) ([]apiv1.Resource, *apiv1.Patcher, error) {
+	/*
+		生成 Ollama Deployment 资源
+		生成 Ollama Service 资源
+		Patcher 中配置 Ollama 服务路径
+	*/
+	var resources []apiv1.Resource
+
+	// Build Kubernetes Deployment for the local MySQL instance.
+	localDeployment, err := infer.generateDeployment(request)
+	if err != nil {
+		return nil, nil, err
+	}
+	resources = append(resources, *localDeployment)
+
+	// Build Kubernetes Service for the local MySQL instance.
+	localSvc, hostAddress, err := infer.generateService(request)
+	if err != nil {
+		return nil, nil, err
+	}
+	resources = append(resources, *localSvc)
+
+	envVars := []v1.EnvVar{
+		{
+			Name:  "INFERENCE_PATH",
+			Value: hostAddress,
+		},
+	}
+	patcher := &apiv1.Patcher{
+		Environments: envVars,
+	}
+
+	return resources, patcher, nil
+}
+
+// generatePodSpec generates the Kubernetes PodSpec for the Inference instance.
+func (infer *Inference) generatePodSpec(_ *module.GeneratorRequest) (v1.PodSpec, error) {
+	var mountPath string
+	var modelPullCmd []string
+	var containerPort int32
+	switch infer.Framework {
+	case "ollama":
+		mountPath = "/root/.ollama"
+		modelPullCmd = append(modelPullCmd, "ollama", "pull", infer.Framework)
+		containerPort = 11434
+	default:
+	}
+
+	image := OllamaImage
+
+	volumes := []v1.Volume{
+		{
+			Name: infer.Framework + inferStorageSuffix,
+			VolumeSource: v1.VolumeSource{
+				EmptyDir: &v1.EmptyDirVolumeSource{},
+			},
+		},
+	}
+
+	volumeMounts := []v1.VolumeMount{
+		{
+			Name:      infer.Framework + inferStorageSuffix,
+			MountPath: mountPath,
+		},
+	}
+
+	ports := []v1.ContainerPort{
+		{
+			Name:          infer.Framework,
+			ContainerPort: containerPort,
+		},
+	}
+
+	podSpec := v1.PodSpec{
+		Containers: []v1.Container{
+			{
+				Name:         infer.Framework,
+				Image:        image,
+				Ports:        ports,
+				Command:      modelPullCmd,
```

Comment:
shouldn't this be `ollama run` instead of `ollama pull`?

Reply:
`ollama run` will start a cli for chat. As long as `pull` one model, we could use it by ollama api.